src/multicore.rs

```diff
@@ -42,17 +40,18 @@ mod implementation {
             log2_floor(*NUM_CPUS)
         }
 
-        pub fn compute<F, R>(&self, f: F) -> WorkerFuture<R::Item, R::Error>
+        pub fn compute<F, R>(&self, f: F) -> Waiter<R>
         where
             F: FnOnce() -> R + Send + 'static,
-            R: IntoFuture + 'static,
-            R::Future: Send + 'static,
-            R::Item: Send + 'static,
-            R::Error: Send + 'static,
+            R: Send + 'static,
         {
-            WorkerFuture {
-                future: CPU_POOL.spawn_fn(f),
-            }
+            let (sender, receiver) = bounded(1);
```

Comment:
It should be 1, otherwise the `send` would block until the receive.

Reply:
Is that bad in this case? The reason I ask is that we have previously seen behavior (elsewhere) where not syncing caused performance problems because the scheduler didn't actually flush the channels to the receiver promptly. I think in that case extra resources are also consumed. It might (unless you're already sure) be worth giving it a try to see how performance and resource usage are affected. I'm perfectly willing to believe it will hurt but am curious after the previous experience.

Reply:
It means that the threads have to stay alive longer than necessary. I don't have an easy way to test the real perf on a larger thing, but I am pretty sure about this. What scheduler/channels issue are you referring to?